tests/disk_check_test.py

```diff
@@ -177,6 +180,72 @@ def test_readonly(self, mock_proc, mock_log):
             
         assert max_log_lvl == syslog.LOG_ERR
 
+    @patch("disk_check.syslog.syslog")
+    @patch("disk_check.subprocess.run")
+    @patch('os.access', return_value=True)
+    @patch('os.statvfs', return_value=os.statvfs_result((4096, 4096, 1909350, 1491513, 0,
+                                                         971520, 883302, 883302, 4096, 255)))
+    def test_mount_disk_full(self, mock_os_statvfs, mock_os_access, mock_proc, mock_log):
+        global max_log_lvl
+        max_log_lvl = -1
+        mock_proc.side_effect = mock_subproc_run
+        mock_log.side_effect = report_err_msg
+
+        tc = {
+            "upperdir": "/tmp",
+        }
+        swap_upper(tc)
+
+        with patch('sys.argv', ["", "-d", "/tmpx"]):
+            disk_check.main()
+
+    @patch("disk_check.syslog.syslog")
+    @patch("disk_check.subprocess.run")
+    @patch('shutil.rmtree')
+    @patch('os.access', return_value=True)
+    @patch('os.statvfs', return_value=os.statvfs_result((4096, 4096, 1909350, 1491513, 4096,
+                                                         971520, 883302, 883302, 4096, 255)))
+    def test_unmount_disk_full(self, mock_os_statvfs, mock_os_access, mock_rmtree, mock_proc, mock_log):
```

Comment:
>test_unmount_disk_full [](http://example.com/codeflow?start=8&length=22)

As discussed in internal meeting, disk-full is special compared with disk-readonly is that it can recover to normal (delete large files). Does this testcase cover the recovery?

Reply:
Yes, this test will cover recovery case.
I also will add a sonic-mgmt test to cover the disk-full scenario.